fixtures/vitest-pool-workers-examples/rpc/src/index.ts

```diff
@@ -0,0 +1,138 @@
+import { DurableObject, RpcTarget, WorkerEntrypoint } from "cloudflare:workers";
+
+export class Counter extends RpcTarget {
+	#value: number;
+
+	constructor(value: number) {
+		super();
+		this.#value = value;
+	}
+
+	get value() {
+		return this.#value;
+	}
+
+	increment(by = 1) {
+		return (this.#value += by);
+	}
+
+	clone() {
+		return new Counter(this.#value);
+	}
+}
+
+export class TestObject extends DurableObject<Env> {
```

Comment:
What's the state of returning non-primitives over RPC? If it works, it would be great to have a test showing this - if it doesn't, the isolated storage assertion should be updated to mention that objects over DO RPC aren't currently supported

Reply:
I think that objects and functions are generally OK but classes need to derive from `RpcTarget`?

See https://developers.cloudflare.com/workers/runtime-apis/rpc/#structured-clonable-types-and-more

Reply:
When we were looking at the larger PR, there was a bug with returning plain objects over RPC, specifically from durable objects - would be great to know whether this is now supported or needs work

Reply:
Is it possible to get a test showing this?

Reply:
https://github.com/cloudflare/workers-sdk/pull/5916/commits/6e6292de5e0b3e002185d9a41b85745cdd6b7084